src/scripts/main.js

```diff
@@ -1 +1,9 @@
 'use strict';
+
+window.addEventListener('hashchange', () => {
+  if (window.location.hash === '#menu') {
+    document.body.classList.add('page__body--with-menu');
+  } else {
+    document.body.classList.remove('page__body--with-menu');
+  }
```

Comment:
you can use [toggle](https://developer.mozilla.org/en-US/docs/Web/API/DOMTokenList/toggle) function here instead of `if else` statement